scripts/publish-local.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+set -e
+
+if [[ -n "${OSMESA_DEBUG}" ]]; then
+    set -x
+fi
+
+SBT_SCRIPT=https://raw.githubusercontent.com/locationtech/geotrellis/96975b45ed1d32cb3075c30ecb4023bfea540ec1/sbt
+
+function usage() {
+    echo -n "Usage: $(basename "$0") [OPTIONS]
+
+Clones, builds and locally publishes a project
+
+OPTIONS:
+    geotrellis - GeoTrellis master
+    geomesa - GeoMesa master
+    vectorpipe - VectorPipe master
+    all - Build and publish all of the above
+"
+}
+
+if [ "${BASH_SOURCE[0]}" = "${0}" ]
+then
+    if [ "${1:-}" = "--help" ]
+    then
+        usage
+    else
+        case $1 in
+            geotrellis) PUBLISH_GEOTRELLIS=1 ;;
+            geomesa)    PUBLISH_GEOMESA=1 ;;
+            vectorpipe) PUBLISH_VECTORPIPE=1 ;;
+            all)
+                PUBLISH_GEOTRELLIS=1
+                PUBLISH_GEOMESA=1
+                PUBLISH_VECTORPIPE=1
+                ;;
+            help|*)           usage; exit 1 ;;
+        esac
+
+        mkdir -p ./third_party_sources
+        pushd third_party_sources
+
+        if [ -n "$PUBLISH_GEOTRELLIS" ]; then
+            if [ ! -d "geotrellis" ]; then
+                git clone https://github.com/locationtech/geotrellis.git geotrellis
+            fi
+            pushd geotrellis
+
+            git pull
+            ./scripts/publish-local.sh
+
+            popd
+        fi
+
+        if [ -n "$PUBLISH_GEOMESA" ]; then
+            if [ ! -d "geomesa" ]; then
+                git clone https://github.com/locationtech/geomesa.git geomesa
+            fi
+            pushd geomesa
+
+            git pull
+#            mvn clean install -T8 -am -Dmaven.test.skip;
```

Comment:
Getting stuck on this step when running from inside the VM. Initially, things would lock up. I believe I got around that by reducing the number of threads (`-T`) to 2 (along with the directory location change above).

Now trying to get past:

```
[ERROR] Failed to execute goal on project geomesa-accumulo-jobs_2.11: Could not resolve dependencies for project org.locationtech.geomesa:geomesa-accumulo-jobs_2.11:jar:1.4.0-SNAPSHOT: Failure to find org.locationtech.geomesa:geomesa-accumulo-datastore_2.11:jar:tests:1.4.0-SNAPSHOT in https://repo.locationtech.org/content/groups/snapshots was cached in the local repository, resolution will not be reattempted until the update interval of locationtech-snapshots has elapsed or updates are forced -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :geomesa-accumulo-jobs_2.11
```

Reply:
We should skip the `clean` to speed up incremental builds.

`-T2.0C` might also be better to support machines with different numbers of cores (assuming that that's not the source of the build failure, something I've also run into with a different manifesting project).

(My build is still running, but I'm hopeful that I've been able to get past it by not skipping tests, since that seems to be a dependency it's complaining about.)

Reply:
I'm getting a variety of errors, including "The forked VM terminated without properly saying goodbye. VM crash or System.exit called?"

Building outside of the VM is probably preferable for various reasons (and/or providing downloadable artifacts for manually installing into `~/.m2`).